File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/Python/s580342216_commented.py
import sys

class Solution:
    def __init__(self):
        for line in sys.stdin:
            line = line.strip()
            if not line:
                continue
            self.run_case(line)
    
    def run_case(self, line):
        parts = line.split()
        N = int(parts[0])
        K = int(parts[1])
        length = 0
        n = N
        while n > 0:
            n = n // K
            length += 1
        print(length)

class LCS:
    def lcs(self, A, B):
        m = len(A)
        n = len(B)
        self.dp = [[0]*(n+1) for _ in range(m+1)]
        for i in range(1, m+1):
            for j in range(1, n+1):
                if A[i-1] == B[j-1]:
                    self.dp[i][j] = self.dp[i-1][j-1] + 1
                else:
                    self.dp[i][j] = max(self.dp[i-1][j], self.dp[i][j-1])
        return self.dp[m][n]
    
    def getLcs(self, X, Y):
        m = len(X)
        n = len(Y)
        index = self.dp[m][n]
        lcs = []
        i = m
        j = n
        while i > 0 and j > 0:
            if X[i-1] == Y[j-1]:
                lcs.append(X[i-1])
                i -= 1
                j -= 1
            else:
                if self.dp[i-1][j] > self.dp[i][j-1]:
                    i -= 1
                else:
                    j -= 1
        lcs.reverse()
        return ''.join(lcs)

if __name__ == "__main__":
    Solution()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/comment/direct/Java/C++/s580342216_commented.cpp
#include <iostream>
#include <sstream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    Solution() {
        string line;
        while (getline(cin, line)) {
            run_case(line);
        }
    }

private:
    void run_case(const string& line) {
        istringstream iss(line);
        int N, K;
        iss >> N >> K;

        int len = 0;
        while (N > 0) {
            N /= K;
            len++;
        }
        cout << len << endl;
    }

    vector<int> strToIntArray(const string& str) {
        istringstream iss(str);
        vector<int> res;
        int num;
        while (iss >> num) {
            res.push_back(num);
        }
        return res;
    }
};

class LCS {
public:
    vector<vector<int>> dp;

    int lcs(const string& A, const string& B) {
        int sz_a = A.length(), sz_b = B.length();
        dp.assign(sz_a + 1, vector<int>(sz_b + 1, 0));

        for (int i = 0; i <= sz_a; ++i) dp[i][0] = 0;
        for (int j = 0; j <= sz_b; ++j) dp[0][j] = 0;

        for (int i = 1; i <= sz_a; ++i) {
            for (int j = 1; j <= sz_b; ++j) {
                if (A[i-1] == B[j-1]) {
                    dp[i][j] = dp[i-1][j-1] + 1;
                } else {
                    dp[i][j] = max(dp[i-1][j], dp[i][j-1]);
                }
            }
        }
        return dp[sz_a][sz_b];
    }

    string getLCS(const string& X, const string& Y) {
        int m = X.length(), n = Y.length();
        int index = dp[m][n];
        int temp = index;

        string lcs_str;
        lcs_str.reserve(index);

        int i = m, j = n;
        while (i > 0 && j > 0) {
            if (X[i-1] == Y[j-1]) {
                lcs_str.push_back(X[i-1]);
                i--;
                j--;
            } else if (dp[i-1][j] > dp[i][j-1]) {
                i--;
            } else {
                j--;
            }
        }

        reverse(lcs_str.begin(), lcs_str.end());
        return lcs_str;
    }
};

int main() {
    Solution sol;
    return 0;
}
// End of Code
